File: sdk/clojure/src/hitechcloud/sdk/qr_and_thanh_toan.clj
(ns hitechcloud.sdk.qr_and_thanh_toan
  (:require [hitechcloud.sdk.http-client :as http]))

;; QR & Thanh toán

(defn banks "GET /api/banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã..." [client & [params]] (http-get client "/api/banks" params))

(defn banks_2 "GET /api/banks/2 - Tra một ngân hàng theo mã BIN (970436)" [client & [params]] (http-get client "/api/banks/2" params))

(defn generator_vietqr "GET /api/generator/vietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển..." [client & [params]] (http-get client "/api/generator/vietqr" params))

(defn tools_qr_generate "POST /api/tools/qr/generate - Sinh mã QR từ nội dung bất kỳ" [client & [params]] (http-post client "/api/tools/qr/generate" params))

(defn tools_qr_vcard "POST /api/tools/qr/vcard - Sinh mã QR danh thiếp vCard" [client & [params]] (http-post client "/api/tools/qr/vcard" params))

File: sdk/fsharp/src/HiTechCloudSDK/QrAndThanhToan.fs
namespace HiTechCloud.SDK

/// QR & Thanh toán
type QrAndThanhToanResource(httpClient: HttpClient) =

    /// GET /api/banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã...
    member this.Banks() =
        async {
            return! httpClient.getAsync("/api/banks")
        }

    /// GET /api/banks/2 - Tra một ngân hàng theo mã BIN (970436)
    member this.Banks2() =
        async {
            return! httpClient.getAsync("/api/banks/2")
        }

    /// GET /api/generator/vietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển...
    member this.GeneratorVietqr() =
        async {
            return! httpClient.getAsync("/api/generator/vietqr")
        }

    /// POST /api/tools/qr/generate - Sinh mã QR từ nội dung bất kỳ
    member this.ToolsQrGenerate() =
        async {
            return! httpClient.postAsync("/api/tools/qr/generate")
        }

    /// POST /api/tools/qr/vcard - Sinh mã QR danh thiếp vCard
    member this.ToolsQrVcard() =
        async {
            return! httpClient.postAsync("/api/tools/qr/vcard")
        }

